src/core/org/apache/jmeter/JMeter.java

```diff
@@ -997,6 +997,7 @@ private void runNonGui(String testFile, String logFile, boolean remoteStart, Str
             File f = new File(testFile);
             if (!f.exists() || !f.isFile()) {
                 println("Could not open " + testFile);
+                System.exit(1);
```

Comment:
While the bug report is about JMeter not exiting with a non zero value, I think we should throw an exception here (my preferred one would be `ConfigurationException`). That makes it easier to test -- as you have noticed in your tests -- and nicer for other people, that use this part as a library.